api/v1/helpers/verbs/utils.js

```diff
@@ -237,9 +237,36 @@ function handleAssociations(reqObj, inst, props, method) {
   });
 } // handleAssociations
 
-function isMultipleAssociations(opts) {
-  const MULTIPLE_ASSOCIATIONS = 3;
-  return opts.attributes.length >= MULTIPLE_ASSOCIATIONS;
+/**
+ * When more than one association is required, Sequelize creates inner query
+ * but doesn't expose FK causing SQL failure.
+ *
+ * This method, in order to avoid Sequelize issue, will add all
+ * the foreign keys to the query when more than one association is required.
+ */
+function multipleAssociations(helperModule, requestedAttributes) {
+  // Filters all association just when requested
+  const associations = requestedAttributes
+    .filter((attribute) => helperModule.model.associations[attribute]);
+
+  if (associations.length > 1) {
+    /*
+     Maps association names (User, currentCollector, etc) to model.fk_names
+     (createdBy, collectorId, etc)
+     */
+    const extraForeignKeys = associations
+      .map((assocName) => {
+        const foreignKey = helperModule.model.associations[assocName]
+          .foreignKey;
+        const attribute = helperModule.model.attributes[foreignKey];
+        return attribute ? attribute.fieldName : '';
+      })
+      .filter((it) => it !== '');
```

Comment:
I think it would be better to combine this check with line 262.

Reply:
I am not sure if I got your point. 

Problem here is that a association doesnt have the FK in the model and there is no way to skip a .map iteration when that happens.

I have changed a bit doing a filter first getting just associations with FK in the model and then the .map will not be affected, will be 1:1 (association fk to model attribute fk).

Same result and number of iterations... maybe easier to read?!

Reply:
You're right, makes sense.